wordpress-mobile/WordPress-Android
# Issue/reflect changes in threaded comments in post details

This PR fixes the issue with comment numbers not updating in Reader Post Details after new comment have been added in the post comment list.

Number of comments in bottom bar and content of the comment snippet should be reloaded from local cache after you navigate from comment list. I did not attach the refresh logic to any particular action performed in comment list, since a lot of actions there can require reloading data in post details (loading more comments with PTR, adding new comment, moderating, comment editing).

I introduced a couple of changes intended to make the comment snippet logic a bit more testable. 

Comment snipped supposed to be updated if it's in a normal state (eg, not loading or failing). I'm not really sure how to test this this.

To test:
- Open the post details of the post with no comments, and navigate to comment list.
- Leave a comment or two and navigate back to post details.
- Confirm that the comment snippet shows the first comment you left, and the counter at bottom bar and in comment snippet reflects the number of comments you have added.

## Regression Notes
1. Potential unintended areas of impact
Potential unforeseen issues when reloading data from local cache.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and existing test.

3. What automated tests I added (or what prevented me from doing so)
Added a test for local cache reload.

PR submission checklist:
